### PR TITLE
Added partial support to Dynamic type

### PR DESCRIPTION
**Pull request by nikordaris**

### CODE REVIEW

File: graphene/types/dynamic.py [hunk 1]
```diff
@@ -11,7 +12,7 @@ class Dynamic(MountedType):
 
     def __init__(self, type, with_schema=False, _creation_counter=None):
         super(Dynamic, self).__init__(_creation_counter=_creation_counter)
-        assert inspect.isfunction(type)
+        assert inspect.isfunction(type) or isinstance(type, partial)
```
Comment (jkimbo) [inline graphene/types/dynamic.py]: I think a better check would be to use `callable` then it would catch both cases.

Comment (nikordaris) [reply graphene/types/dynamic.py]: You can't use callable because class types are callable.

Comment (nikordaris) [reply graphene/types/dynamic.py]: I could do ```not inspect.isclass(type) and callable(type)```

Comment (nikordaris) [reply graphene/types/dynamic.py]: Really though I was basing it off of https://github.com/graphql-python/graphene/blob/12d4dab7742c9408777380c390152d7cef0f838d/graphene/types/utils.py#L44

I wasn't sure if there were callable types we didn't want to support

Comment (jkimbo) [reply graphene/types/dynamic.py]: Ah yeah you're right. What you've done then looks good. Just needs some tests.

**Comment by nikordaris** [comment]
ok @jkimbo I've added a test for partial. let me know if there is anything else i should change

**Comment by syrusakbary** [comment]
It goods look to merge for me @jkimbo, feel free to merge once you review it :)

**Comment by jkimbo** [comment]
@syrusakbary I can't merge, I need write access to do that.

**Comment by jkimbo** [comment]
This has been released in v2.1.1

**Pull request merged**